addons/controllers/addon_controller.go

```diff
@@ -315,6 +333,19 @@ func (r *AddonReconciler) reconcileNormal(
 	return result, nil
 }
 
+func (r *AddonReconciler) addFinalizer(ctx context.Context, o, deepCopy client.Object) error {
+	controllerutil.AddFinalizer(deepCopy, addontypes.AddonFinalizer)
```

Comment:
nit: Should `if controllerutil.ContainsFinalizer(deepCopy, addontypes.AddonFinalizer)` be added here similar to removeFinalizer ? Or should this check be removed from removeFinalizer ?

Reply:
you are correct, we should check to see if it is already there and do nothing if so

Reply:
done